tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py

```diff
@@ -742,3 +742,83 @@ def test_connectivity_field_inverse_image_non_contiguous():
 
     with pytest.raises(ValueError, match="generates non-contiguous dimensions"):
         e2v_conn.inverse_image(UnitRange(V_START, V_STOP))
+
+
+def test_connectivity_field_inverse_image_2d_domain_skip_values():
+    V = Dimension("V")
+    E = Dimension("E")
+    E2V = Dimension("E2V")
+
+    V_START, V_STOP = 0, 3
+    E_START, E_STOP = 0, 4
+    E2V_START, E2V_STOP = 0, 4
+
+    e2v_conn = common._connectivity(
+        np.asarray([[-1, 0, 2, -1], [1, 1, 2, 2], [2, 2, -1, -1], [-1, 2, -1, -1]]),
+        domain=common.domain(
+            [
+                common.named_range((E, (E_START, E_STOP))),
+                common.named_range((E2V, (E2V_START, E2V_STOP))),
+            ]
+        ),
+        codomain=V,
+    )
+
+    # e2v_conn:
+    #  ---E2V---------
+    #  |[[-1  0  2 -1]
+    #  E [ 1  1  2  2]
+    #  | [ 2  2 -1 -1]
+    #  | [-1  2 -1 -1]]
+
+    image_range = UnitRange(V_START, V_STOP)
+    result = e2v_conn.inverse_image(image_range)
+
+    assert len(result) == 2
+    assert result[0] == (E, UnitRange(E_START, E_STOP))
+    assert result[1] == (E2V, UnitRange(E2V_START, E2V_STOP))
+
+    result = e2v_conn.inverse_image(UnitRange(0, 2))
+    assert len(result) == 2
+    assert result[0] == (E, UnitRange(0, 2))
+    assert result[1] == (E2V, UnitRange(0, 2))
+
+    result = e2v_conn.inverse_image(UnitRange(0, 1))
+    assert len(result) == 2
+    assert result[0] == (E, UnitRange(0, 1))
+    assert result[1] == (E2V, UnitRange(1, 2))
+
+    result = e2v_conn.inverse_image(UnitRange(1, 2))
+    assert len(result) == 2
+    assert result[0] == (E, UnitRange(1, 2))
+    assert result[1] == (E2V, UnitRange(0, 2))
+
+    with pytest.raises(ValueError, match="generates non-contiguous dimensions"):
+        result = e2v_conn.inverse_image(UnitRange(1, 3))
+
+    with pytest.raises(ValueError, match="generates non-contiguous dimensions"):
+        result = e2v_conn.inverse_image(UnitRange(2, 3))
+
+
+@pytest.mark.parametrize(
+    "select, ignore_mask, expected",
```

Comment:
```suggestion
    "select, ignore_value, expected",
```

Reply:
for this function not really, let's discuss after my next push